consensus-types/blocks/kzg.go

```diff
@@ -80,6 +86,67 @@ func MerkleProofKZGCommitment(body interfaces.ReadOnlyBeaconBlockBody, index int
 	return proof, nil
 }
 
+// PrecomputeMerkleProofComponents pre-computes the expensive parts of Merkle proof generation
+// that are shared across all blob indices for a given block body.
+func PrecomputeMerkleProofComponents(body interfaces.ReadOnlyBeaconBlockBody) (*MerkleProofComponents, error) {
```

Comment:
shouldn't we have unit tests for these? or is it covered as part of the bench

Reply:
so this was covered in `TestServer_buildBlobSidecars` and in `TestProposer_ProposeBlock_OK`, but I can add individual unit tests as well if that's what we want